docs/workflow-rbac.md

```diff
@@ -48,3 +48,28 @@ Warning: For many organizations, it may not be acceptable to give a workflow the
 
 If you are not using the emissary, you'll need additional permissions.
 See [executor](https://github.com/argoproj/argo-workflows/tree/main/manifests/quick-start/base/executor) for suitable permissions.
+
+**Note**: Prior to `v3.5.2`, simple workflows such as [hello-world](https://raw.githubusercontent.com/argoproj/argo-workflows/main/examples/hello-world.yaml) may have run successfully without RBAC. As of `v3.5.2`, these workflows will fail without at least the permissions above via associated [Service Accounts](service-accounts.md). For example:
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: hello-world
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: hello-world-executor
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: executor
+subjects:
+- kind: ServiceAccount
+  name: hello-world
+```
+
+```bash
+argo submit --serviceaccount hello-world --watch https://raw.githubusercontent.com/argoproj/argo-workflows/main/examples/hello-world.yaml
+```
```

Comment:
I don't think this is necessary. The docs have never said otherwise that Workflow RBAC was not required, that was an incidental technicality that simple steps could work without it.
It's something we can mention in upgrade notes, not in this page. Also, realistically it only impacts a small percentage of production users.

Additionally
1. SAs are already mentioned at the top of this page
1. The example can be useful to demonstrate combining everything in this page, but for a footnote, this makes it seem like required reading for everyone, even though it likely will not impact them. A footnote shouldn't need an example.
1. Versioning is specified with `> v3.5.2 and after` per #12581 
1. Should be simpler and more direct, per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)

Reply:
Addressed in this [follow-up PR](https://github.com/argoproj/argo-workflows/pull/12686)